bots/inspect-queue

```diff
@@ -21,38 +21,57 @@ MAX_PRIORITY = 9
 
 import argparse
 import sys
-import time
+import ssl
 
-import amqp
+import pika
+
+seen_messages = 0
 
 def main():
     parser = argparse.ArgumentParser(description='Read and print messages from the queue without acknowleding them')
     parser.add_argument('--amqp', default='localhost',
                         help='The URL of the AMQP server to consume from')
     parser.add_argument('--queue', default='tasks',
                         help='The name of the queue to consume from')
-    parser.add_argument('-n', '--number', default=0,
-                        help='Number of queue items to show, starting from the front (0=all)')
-    parser.add_argument('-t', '--timeout', default=0.5,
-                        help='Time to wait for messages')
     opts = parser.parse_args()
 
-    connection = amqp.Connection(host=opts.amqp)
-    connection.connect()
+    host, port = opts.amqp.split(':')
+    connection = pika.BlockingConnection(pika.ConnectionParameters(
+        host=host,
+        port=int(port),
+        ssl=True,
+        ssl_options=pika.SSLOptions(
+            ssl_version=ssl.PROTOCOL_TLSv1_2,
+            cafile='/run/secrets/webhook/ca.pem',
+            keyfile='/run/secrets/webhook/amqp-client.key',
+            certfile='/run/secrets/webhook/amqp-client.pem',
+            server_hostname='amqp-cockpit.apps.ci.centos.org'),
+        credentials=pika.credentials.ExternalCredentials()))
     channel = connection.channel()
     try:
-        channel.queue_declare(queue=opts.queue, passive=True, auto_delete=False)
-    except amqp.exceptions.NotFound:
+        declare_result = channel.queue_declare(queue=opts.queue, passive=True, auto_delete=False)
+    except pika.exceptions.ChannelClosed:
+        sys.stdout.write('queue is empty\n')
+        return 0
+    count = declare_result.method.message_count
+    if count == 0:
         sys.stdout.write('queue is empty\n')
         return 0
-    channel.basic_qos(0, opts.number, True)
+    channel.basic_qos(0, count, True)
 
-    def callback(msg):
-        sys.stdout.write("{0}\n".format(msg.body))
-        msg.channel.basic_reject(msg.delivery_tag, requeue=True)
+    def callback(channel, method_frame, header_frame, body):
+        global seen_messages
+        sys.stdout.write("{0}\n".format(body))
+        seen_messages += 1
+        if seen_messages == count:
+            channel.stop_consuming()
 
-    channel.basic_consume(callback=callback, queue=opts.queue)
-    time.sleep(opts.timeout)
+    channel.basic_consume(callback, queue=opts.queue)
+    sys.stdout.write('Ctrl+C to stop consuming\n')
+    try:
+        channel.start_consuming()
+    except KeyboardInterrupt:
+        channel.stop_consuming()
```

Comment:
This is just a debugging tool, but doesn't pika have a way to consume an item in a non-blocking way, and return None or so if there's no item? Or specify a timeout? In the worst case, we could create a timeout ourselves (`signal.alarm()`). However, this is fine as a follow-up.

Reply:
while it doesn't seem like it, inspect-queue will consume the exact amount of messages here that are in the queue, and then quit. You can just also quit via ctrl+c.

I'll try with basic_get, to see if that yields clearer code.